lib/distro.py

```diff
@@ -43,7 +43,7 @@ def __init__(self, name=None, version=None, architecture=None):
                 distribution
         """
         self.lsb_name = name
-        if architecture.upper() not in SUPPORTED_ARCHITECTURES:
+        if architecture.lower() not in SUPPORTED_ARCHITECTURES:
```

Comment:
Using lower or upper case is a random personal choice, both work. Thus, I see no problem in this.